goc/app.py

```diff
@@ -45,17 +54,86 @@ def blog():
     }
     if(blog_id == '3434'):
         return render_template('blog.j2', title = blog['title'], blog = blog)
-    else : 
+    else :
         return 'Error'
 
 @app.route('/login')
-def login(): 
+def login():
     pass
 
 @app.route('/signup')
-def signup(): 
+def signup():
     pass
 
+@app.route('/submitBlog',methods = ['POST','GET'])
+def submitBlog():    
+    shortlisted = 0
+    blog_form  = BlogForm()   
+    if(blog_form.isSelected.data):
+        shortlisted = 1
+        blog_form.interview.interview_rounds.append_entry()
+        return render_template('blogform.j2',blog_form = blog_form,shortlisted = shortlisted)
+    if(blog_form.addInterview.data):
+        blog_form.interview.interview_rounds.append_entry()        
+        return render_template('blogform.j2',blog_form = blog_form,shortlisted = 1)
+    if(blog_form.addShortListing.data):
+        blog_form.shortlisting.shortlisting_rounds.append_entry()
+        if(len(blog_form.interview.interview_rounds)>0):
+            return render_template('blogform.j2',blog_form = blog_form,shortlisted = 1)
+        else:
+            return render_template('blogform.j2',blog_form = blog_form,shortlisted = 0)        
+    if(blog_form.addTag.data):
+        blog_form.tags.append_entry()
+        if(len(blog_form.interview.interview_rounds)>0):
+            return render_template('blogform.j2',blog_form = blog_form,shortlisted = 1)
+        else:
+            return render_template('blogform.j2',blog_form = blog_form,shortlisted = 0)
+    if(blog_form.validate_on_submit()):       
+        #First make all tags unique
+        tags = [str(x) for x in set(blog_form.tags)]   
+        author = Author(name = str(blog_form.author))                   
+        db.session.add(author)   
```

Comment:
You should put the entire db changes inside try except block to handle unexpected errors. If while adding data to the db raises an error, you can return an error through the except block

Reply:
Done it,Thanks